### PR TITLE
#1110: Make Standardization and Conformance use Spline 0.4.x

### PR DESCRIPTION
**Pull request by benedeki**
* Change Spline version
* Make **Standardization** and **Confluence** depend and init the new Spline module instead of the old ones
* Replacing Spline properties in templates
* Using a different JSON library (instead of the one transitionally ingested from old Spline)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,12 +62,12 @@ The coverage reports are written in each module's `target` directory and aggrega
...
 - [**MongoDB 4.0** installation](https://docs.mongodb.com/manual/administration/install-community/)
-- [**Spline UI deployment**](https://absaoss.github.io/spline/) - place the [spline.war](https://search.maven.org/remotecontent?filepath=za/co/absa/spline/spline-web/0.3.9/spline-web-0.3.9.war)
- in your Tomcat webapps directory (rename after downloading to _spline.war_); NB! don't forget to set up the `spline.mongodb.url` configuration for the _war_
+- [**Spline service deployment**](https://absaoss.github.io/spline/#get-spline)
 - **HADOOP_CONF_DIR** environment variable, pointing to the location of your hadoop configuration (pointing to a hadoop installation)
 
-The _Spline UI_ can be omitted; in such case the **Menas** `za.co.absa.enceladus.spline.urlTemplate` setting should be set to empty string. 
-
+The _Spline service_ can be omitted; in such case the **Standardization** and **Conformance** `spline.producer.url` setting
+ so as **Menas** `lineageReadApiUrl` and `menas.oozie.lineageWriteApiUrl` settings should be all set to empty string.
```
Comment (Zejnilovic) [inline README.md]: I am not sure the `so as` makes sense to me here. `as well as`?

Comment (benedeki) [reply README.md]: `as well as` it is 👍

File: conformance/src/main/resources/spline.properties.template [hunk 2]
```diff
@@ -12,19 +13,13 @@
...
 #
-# Set of properties for setting up persistence to MongoDB.
 #
-spline.persistence.factory=za.co.absa.spline.persistence.api.composition.ParallelCompositeFactory
-spline.persistence.composition.factories=za.co.absa.spline.persistence.mongo.MongoPersistenceFactory,za.co.absa.spline.persistence.hdfs.HdfsPersistenceFactory
-
-spline.mongodb.url=mongodb://localhost:27017
-spline.mongodb.name=spline
-
+# Spline mode, possible values: DISABLED|REQUIRED|BEST_EFFORT (default is BEST_EFFORT)
```
Comment (benedeki) [inline conformance/src/main/resources/spline.properties.template]: I will add better explanation what `spline.mode` means.

File: utils/src/test/scala/za/co/absa/enceladus/utils/validation/field/ScalarFieldValidatorSuite.scala [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package za.co.absa.enceladus.utils.validation.field
+
+import org.apache.spark.sql.types.{MetadataBuilder, StringType, StructField}
+import org.scalatest.FunSuite
+import za.co.absa.enceladus.utils.schema.MetadataKeys
+import za.co.absa.enceladus.utils.types.{Defaults, GlobalDefaults, TypedStructField}
+import za.co.absa.enceladus.utils.validation.ValidationError
+
+class ScalarFieldValidatorSuite extends FunSuite {
```
Comment (GeorgiChochov) [inline utils/src/test/scala/za/co/absa/enceladus/utils/validation/field/ScalarFieldValidatorSuite.scala]: This seems unrelated to the PR, but I'm glad we have more tests :D

Comment (benedeki) [reply utils/src/test/scala/za/co/absa/enceladus/utils/validation/field/ScalarFieldValidatorSuite.scala]: Wonder how it got here. I wasn't explicitly including it, for sure. Some effect of multiple rebases, maybe.
I will check if it's valid and based on that decide on the test suite fate :-)

Comment (benedeki) [reply utils/src/test/scala/za/co/absa/enceladus/utils/validation/field/ScalarFieldValidatorSuite.scala]: Yup, a valid test. It was just missing the Copyright header (similarly a few other tests in the folder). Keeping it.
Its appearance is still a mystery though :-D

Comment (GeorgiChochov) [reply utils/src/test/scala/za/co/absa/enceladus/utils/validation/field/ScalarFieldValidatorSuite.scala]: All good then.

**Pull request merged**